domain/src/main/kotlin/io/github/wykopmobilny/domain/styles/GetAppStyleQuery.kt

```diff
@@ -1,33 +1,47 @@
 package io.github.wykopmobilny.domain.styles
 
+import io.github.wykopmobilny.domain.navigation.NightModeState
+import io.github.wykopmobilny.domain.navigation.SystemSettingsDetector
 import io.github.wykopmobilny.domain.settings.prefs.GetAppearanceSectionPreferences
 import io.github.wykopmobilny.styles.AppThemeUi
 import io.github.wykopmobilny.styles.GetAppStyle
 import io.github.wykopmobilny.styles.StyleUi
-import kotlinx.coroutines.flow.combine
 import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
 import javax.inject.Inject
 
 internal class GetAppStyleQuery @Inject constructor(
-    private val getAppTheme: GetAppTheme,
     private val getAppearanceSectionPreferences: GetAppearanceSectionPreferences,
+    private val systemSettingsDetector: SystemSettingsDetector,
 ) : GetAppStyle {
 
     override fun invoke() =
-        combine(
-            getAppTheme(),
-            getAppearanceSectionPreferences(),
-        ) { theme, appearance ->
-            StyleUi(
-                theme = theme.toUi(),
-                edgeSlidingBehaviorEnabled = !appearance.disableEdgeSlide,
-            )
+        getAppearanceSectionPreferences()
+            .map { appearance ->
+                StyleUi(
+                    theme = mapTheme(appearance.appTheme, appearance.isAmoledTheme),
+                    edgeSlidingBehaviorEnabled = !appearance.disableEdgeSlide,
+                )
+            }.distinctUntilChanged()
+
+    private suspend fun mapTheme(appTheme: AppTheme, amoledTheme: Boolean) =
```

Comment:
🎨 
The variable is initially called `isAmoledTheme`, then it is named `amoledTheme` and at the end it becomes `useAmoledTheme`. I would keep it consistent